app/models/manageiq/providers/base_manager/metrics_capture.rb

```diff
@@ -80,7 +80,7 @@ def filter_perf_capture_now(targets, target_options)
   end
 
   def capture_ems_targets(options = {})
-    Metric::Targets.capture_ems_targets(ems, options)
+    raise(NotImplementedError, _("must be implemented in subclass"))
```

Comment:
Aha.  Here's my understanding:

`perf_capture_object` instantiates `self.class::MetricsCapture`, and eventually `.capture_ems_targets` is called on that object.  Before all the parent fixes like https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/353, this always led here, to `BaseManager::MetricsCapture#capture_ems_targets`.
The more specific implementations like `ContainerManager::MetricsCapture#capture_ems_targets` were unused??
But that didn't matter _because the generic one led to Metric::Targets dispatching by type to same methods `capture_{cloud,container,...}_targets` as the specific implementations!

Now after provider PRs, the right `capture_ems_targets` will get called, so manual dispatch is unnecessary, and the code can be moved to type-specific `capture_ems_targets` implementations.

Reply:
@cben spot on.